agent/http/server.go

```diff
@@ -0,0 +1,194 @@
+package http
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	metrics "github.com/armon/go-metrics"
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-msgpack/codec"
+	"github.com/hashicorp/nomad-autoscaler/agent/config"
+)
+
+const (
+	// healthRoutePattern is the Autoscaler HTTP router pattern which is used
+	// to register the health server endpoint.
+	healthRoutePattern = "/v1/health"
+
+	// metricsRoutePattern is the Autoscaler HTTP router pattern which is used
+	// to register the metrics server endpoint.
+	metricsRoutePattern = "/v1/metrics"
+
+	// healthAliveness is used to define the health of the Autoscaler agent. It
```

Comment:
👍 
i think later on, we'll want a more sophisticated health check for use with consul service discovery, something along the lines of "can I talk to Nomad?"